### PR TITLE
test(MM-48845) : Test cases for channel moveTo category sub menu options

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
Adds test cases for channel move to menu options under

#### Is the feature in stable branch or under review?
- [X] Feature in stable branch (master, main)?
- [ ] Feature still under review? If yes, list all related pull requests for reference.

#### Test server
https://mattermost.atlassian.net/browse/MM-48845

#### Test client

- [X] Browser
- [ ] Mobile App (version, OS)
- [X] Desktop App (version, OS)

<a href="https://gitpod.io/#https://github.com/mattermost/mattermost-test-management/pull/53"></a>

### CODE REVIEW

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Create a custom category in the sidebar
+1. Click on the channel menu option of any channel
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Channels" categories are displayed
+1. Go to the clicked channel
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Channels" categories are displayed
+
+**Step 2**
```
Comment (M-ZubairAhmed) [inline data/test-cases/channels/channel-menu/MM-XXX.md:46]: These are test cases like MM-XX_1, MM-XX_2

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on a GM channel menu option
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+1. Go to the clicked GM channel
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+
+**Step 4**
+
+1. Make sure there is at least two channels are present in the "Channels" category
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:68]: Should be "Make sure there are_"

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+1. Select both channels
+1. Click on either of channels menu option
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Channels" categories are displayed and "Direct Messages" category is not displayed
+
+**Step 5**
+
+1. Make sure there is at least two DMs are present in the "Direct Messages" category
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:76]: Should be "Make sure there are_"...

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+1. Select both DMs
+1. Click on either of DMs menu option
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Direct Messages" categories are displayed and "Channels" category is not displayed
+
+**Step 6**
+
+1. Create a custom category in the sidebar
+1. Make sure there is at least a channel, DM and GM are present
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:85]: Make sure there are at least a channel, DM and GM present

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+1. Select all three channels
+1. Click on either of channels menu option
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:87]: Click on either of the channels menu options

**Review by lindy65 — CHANGES_REQUESTED** [review]
Thanks @M-ZubairAhmed - a few grammatical updates

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 2 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 3 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 5 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 4 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Create a custom category in the sidebar
+1. Click on the channel menu option of any channel
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Channels" categories are displayed
+1. Go to the clicked channel
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:42]: Is the "clicked" channel the one that was selected in line 39?  If so, perhaps this step would read better as "Go to the previously selected channel"?

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Channels" categories are displayed
+
+**Step 2**
+
+1. Create a custom category in the sidebar
+1. Click on a DM channel menu option
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:49]: "Click on any DM channel menu option"

To match the "any" in line 39

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+1. Go to the clicked DM channel
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:52]: Perhaps change to "Go to the previously selected DM channel"

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+
+**Step 3**
+
+1. Create a custom category in the sidebar
+1. Click on a GM channel menu option
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:59]: "Click on any GM channel menu option"

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+1. Go to the clicked GM channel
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:62]: "Go to the previously selected GM channel"

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+
+**Step 4**
+
+1. Make sure there are at least two channels are present in the "Channels" category
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:68]: delete "are" before "present in_"...

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+1. Select both channels
+1. Click on either of channels menu option
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:70]: "Click on either of the channel's menu options"

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Channels" categories are displayed and "Direct Messages" category is not displayed
+
+**Step 5**
+
+1. Make sure there are at least two DMs are present in the "Direct Messages" category
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:76]: Delete "are" before "present in_"...

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+1. Select both DMs
+1. Click on either of DMs menu option
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:78]: "Click on either of the DM's menu options"

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Direct Messages" categories are displayed and "Channels" category is not displayed
+
+**Step 6**
+
+1. Create a custom category in the sidebar
+1. Make sure there are at least a channel, DM and GM present
+1. Select all three channels
+1. Click on either of the channels menu options
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:87]: Click on any one of the channel's menu options

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+1. Make sure there are at least two DMs are present in the "Direct Messages" category
+1. Select both DMs
+1. Click on either of DMs menu option
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Direct Messages" categories are displayed and "Channels" category is not displayed
+
+**Step 6**
+
+1. Create a custom category in the sidebar
+1. Make sure there are at least a channel, DM and GM present
```
Comment (lindy65) [inline data/test-cases/channels/channel-menu/MM-XXX.md:85]: Make sure there is a channel, DM and GM present

**Review by lindy65 — CHANGES_REQUESTED** [review]
Thanks @M-ZubairAhmed - a few grammatical suggestions

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 6 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:42]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 7 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:49]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 9 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:59]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 11 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:68]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 12 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:70]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 16 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:85]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 15 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:87]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 14 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:78]: done

File: data/test-cases/channels/channel-menu/MM-XXX.md (hunk 13 above)
Comment (M-ZubairAhmed) [reply data/test-cases/channels/channel-menu/MM-XXX.md:76]: done

**Review by lindy65 — APPROVED** [review]
Thanks @M-ZubairAhmed - looks good to me 👍

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
+
+**Step 4**
+
+1. Make sure there are at least two channels present in the "Channels" category
+1. Select both channels
+1. Click on either of the channel's menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Channels" categories are displayed and "Direct Messages" category is not displayed
+
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:73]:
```suggestion

**Expected**

For multi-select group of channels, "Move to " sub menu options are  "Favorites" category, followed by "Channels" category and option to create and move to "New Category". "Direct Messages" should not be available.

---
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+**Step 5**
+
+1. Make sure there are at least two DMs present in the "Direct Messages" category
+1. Select both DMs
+1. Click on either of the DM's menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Direct Messages" categories are displayed and "Channels" category is not displayed
+
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:81]:
```suggestion

**Expected**

For multi-select DM group, "Move to " sub menu options are  "Favorites" category, followed by "Direct Messages" category and option to create and move to "New Category". "Channels" category is not available.

---
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+**Step 6**
+
+1. Create a custom category in the sidebar
+1. Make sure there are at least a channel, DM and GM present
+1. Select all three channels
+1. Click on either of the channels menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category category is displayed
+1. Verify that "Channels" and "Direct Messages" categories are not displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:90]:
```suggestion
1. Verify that "Channels" and "Direct Messages" categories are not displayed

**Expected**

For multi-select group consisting of DMs, GMs and channels,  "Move to " sub menu options are  "Favorites" category, followed by created custom category and  option to create and move to "New Category". "Channels" and "Direct Messages" categories are not available.

---
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+**Step 4**
+
+1. Make sure there are at least two channels present in the "Channels" category
+1. Select both channels
+1. Click on either of the channel's menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Channels" categories are displayed and "Direct Messages" category is not displayed
+
+**Step 5**
+
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:75]:
```suggestion

"Move to" sub menu options for multi-select DM group

```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Create a custom category in the sidebar
+1. Click on the channel menu option of any channel
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Channels" categories are displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:41]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category and "Channels" categories are displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+1. Go to the previously selected channel
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Channels" categories are displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:44]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category and "Channels" categories are displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 23]
```diff
@@ -0,0 +1,90 @@
...
+
+**Step 2**
+
+1. Create a custom category in the sidebar
+1. Click on any DM channel menu option
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:51]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category and "Direct Messages" categories are displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 24]
```diff
@@ -0,0 +1,90 @@
...
+1. Go to the previously selected DM channel
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:54]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category and "Direct Messages" categories are displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 25]
```diff
@@ -0,0 +1,90 @@
...
+
+**Step 3**
+
+1. Create a custom category in the sidebar
+1. Click on any GM channel menu option
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:61]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category and "Direct Messages" categories are displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 26]
```diff
@@ -0,0 +1,90 @@
...
+1. Go to the previously selected GM channel
+1. Click on the channel header dropdown menu and click on "Move to..."
+1. Verify that "Favorite" category, followed by created custom category and "Direct Messages" categories are displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:64]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category and "Direct Messages" categories are displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+
+**Step 4**
+
+1. Make sure there are at least two channels present in the "Channels" category
+1. Select both channels
+1. Click on either of the channel's menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Channels" categories are displayed and "Direct Messages" category is not displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:72]:
```suggestion
1. Verify that "Favorites" category, followed by "Channels" categories are displayed and "Direct Messages" category is not displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 28]
```diff
@@ -0,0 +1,90 @@
...
+
+**Step 5**
+
+1. Make sure there are at least two DMs present in the "Direct Messages" category
+1. Select both DMs
+1. Click on either of the DM's menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by "Direct Messages" categories are displayed and "Channels" category is not displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:80]:
```suggestion
1. Verify that "Favorites" category, followed by "Direct Messages" categories are displayed and "Channels" category is not displayed
```

File: data/test-cases/channels/channel-menu/MM-XXX.md [hunk 29]
```diff
@@ -0,0 +1,90 @@
...
+
+**Step 6**
+
+1. Create a custom category in the sidebar
+1. Make sure there are at least a channel, DM and GM present
+1. Select all three channels
+1. Click on either of the channels menu options
+1. Click on the "Move to..." menu item
+1. Verify that "Favorite" category, followed by created custom category category is displayed
```
Comment (jgilliam17) [inline data/test-cases/channels/channel-menu/MM-XXX.md:89]:
```suggestion
1. Verify that "Favorites" category, followed by created custom category category is displayed
```

**Review by jgilliam17 — COMMENTED** [review]
Thanks @M-ZubairAhmed 
I added suggestions for each test step to include title and Expected field.
Feel free to edit the text if needed.
Additionally, I noted in expected a common sub-menu option "New Category" was present in all tested scenarios.

**Comment by saturninoabril** [comment]
Thanks @jgilliam17 for the review!

**Review by saturninoabril — APPROVED** [review]
Thanks @M-ZubairAhmed, LGTM!

**Comment by saturninoabril** [comment]
Test case saved as MM-T5385

**Pull request merged**